pkg/morph/client/client.go

```diff
@@ -356,7 +330,13 @@ func (c *Client) GasBalance() (res int64, err error) {
 		return 0, err
 	}
 
-	return c.client.NEP17BalanceOf(gas, c.acc.PrivateKey().GetScriptHash())
+	gasToken := nep17.New(c.rpcActor, gas)
```

Comment:
It seems to me that you're creating this wrapper again and again in the same `*Client`, most likely it can be done once during init and then reused for the whole `Client` lifetime.